### PR TITLE
interfaces-and-networks: Add a security section with spoof-checking

### PR DESCRIPTION
**Pull request by EdDev**
With the introduction on CNAO v0.58, a new `cnv-bridge` CNI plugin is
deployed which can support MAC address spoof-checking.

Relevant information is provided in a new section on how to utilize this
new option.

### CODE REVIEW

File: docs/virtual_machines/interfaces_and_networks.md [hunk 1]
```diff
@@ -897,3 +897,64 @@ name of the provisioned `NetworkAttachmentDefinition`.
 
 > **Note:** VMIs with macvtap interfaces can be migrated, but their MAC
 > addresses **must** be statically set.
+
+## Security
+
+### MAC spoof check
+
+MAC spoofing refers to the ability to generate traffic with an arbitrary source
+mac address.
```
Comment (phoracek) [inline docs/virtual_machines/interfaces_and_networks.md:906]:
```suggestion
MAC address.
```

File: docs/virtual_machines/interfaces_and_networks.md [hunk 2]
```diff
@@ -897,3 +897,64 @@ name of the provisioned `NetworkAttachmentDefinition`.
...
+An attacker may use this option to generate attacks on the network.
+
+In order to protect against such scenarios, it is possible to enable the
+spoof-checking support in the CNI plugins that support it.
```
Comment (phoracek) [inline docs/virtual_machines/interfaces_and_networks.md:910]:
```suggestion
spoof-checking support in CNI plugins that support it. 
```

File: docs/virtual_machines/interfaces_and_networks.md [hunk 3]
```diff
@@ -897,3 +897,64 @@ name of the provisioned `NetworkAttachmentDefinition`.
...
+### MAC spoof check
+
+MAC spoofing refers to the ability to generate traffic with an arbitrary source
+mac address.
+An attacker may use this option to generate attacks on the network.
+
+In order to protect against such scenarios, it is possible to enable the
+spoof-checking support in the CNI plugins that support it. 
+
+The POD primary network which is served by the cluster network provider
```
Comment (phoracek) [inline docs/virtual_machines/interfaces_and_networks.md:912]:
```suggestion
The pod primary network which is served by the cluster network provider
```

File: docs/virtual_machines/interfaces_and_networks.md [hunk 4]
```diff
@@ -897,3 +897,64 @@ name of the provisioned `NetworkAttachmentDefinition`.
...
+is not covered by this documentation. Please refer to the relebant provider to
```
Comment (phoracek) [inline docs/virtual_machines/interfaces_and_networks.md:913]:
```suggestion
is not covered by this documentation. Please refer to the relevant provider to
```

File: docs/virtual_machines/interfaces_and_networks.md [hunk 5]
```diff
@@ -897,3 +897,64 @@ name of the provisioned `NetworkAttachmentDefinition`.
...
+check how to enable spoofing check.
+The following text refers to the secondary networks, served using multus.
+
+There are two known CNI plugins that support mac-spoof-check:
```
Comment (phoracek) [inline docs/virtual_machines/interfaces_and_networks.md:917]: In a paragraph above you used `spoof-checking`. Would you please unify it?

File: docs/virtual_machines/interfaces_and_networks.md [hunk 6]
```diff
@@ -897,3 +897,64 @@ name of the provisioned `NetworkAttachmentDefinition`.
...
+- [sriov-cni](https://github.com/openshift/sriov-cni):
+  Through the `spoofchk` parameter .
+- cnv-bridge: Through the `macspoofchk`.
+
+> **Note:** `cnv-bridge` is provided by
+  [CNAO](https://github.com/kubevirt/cluster-network-addons-operator).
+  The [bridge-cni](https://github.com/containernetworking/plugins) is planned
+  to support the `macspoofchk` options as well.
+
+The configuration is to be done on the NAD by the operator and any interface
```
Comment (phoracek) [inline docs/virtual_machines/interfaces_and_networks.md:928]:
```suggestion
The configuration is to be done on the NetworkAttachmentDefinition by the operator and any interface
```

File: docs/virtual_machines/interfaces_and_networks.md [hunk 7]
```diff
@@ -897,3 +897,64 @@ name of the provisioned `NetworkAttachmentDefinition`.
...
+that points to the NAD will have this feature enabled.
```
Comment (phoracek) [inline docs/virtual_machines/interfaces_and_networks.md:929]:
```suggestion
that points to the NetworkAttachmentDefinition will have this feature enabled.
```

File: docs/virtual_machines/interfaces_and_networks.md [hunk 8]
```diff
@@ -897,3 +897,64 @@ name of the provisioned `NetworkAttachmentDefinition`.
...
+  config: '{
+            "cniVersion": "0.3.1",
+            "name": "br-spoof-check",
+            "type": "cnv-bridge",
+            "bridge": "br10",
+            "macspoofchk": true
+        }'
+```
+
+On the VMI, the network section should point to this NAD by name:
```
Comment (phoracek) [inline docs/virtual_machines/interfaces_and_networks.md:947]:
```suggestion
On the VMI, the network section should point to this NetworkAttachmentDefinition by name:
```

File: docs/virtual_machines/interfaces_and_networks.md [hunk 9]
```diff
@@ -897,3 +897,64 @@ name of the provisioned `NetworkAttachmentDefinition`.
...
+  - name: default
+    pod: {}
+  - multus:
+      networkName: br-spoof-check
+    name: br10
+```
+
+#### Limitations
+
+- The `cnv-bridge` CNI supports mac-spoof-checking through nftables, therefore
```
Comment (phoracek) [inline docs/virtual_machines/interfaces_and_networks.md:959]: Ditto, make `mac-spoof-checking` consistent

**Review by phoracek — CHANGES_REQUESTED** [review]
Looks good overall, just some nits

File: docs/virtual_machines/interfaces_and_networks.md (hunk 7 above)
Comment (EdDev) [reply docs/virtual_machines/interfaces_and_networks.md:929]: Done

File: docs/virtual_machines/interfaces_and_networks.md (hunk 8 above)
Comment (EdDev) [reply docs/virtual_machines/interfaces_and_networks.md:947]: Done

File: docs/virtual_machines/interfaces_and_networks.md (hunk 9 above)
Comment (EdDev) [reply docs/virtual_machines/interfaces_and_networks.md:959]: Done

**Comment by EdDev** [comment]
[change](https://github.com/kubevirt/user-guide/compare/3008c04746a6d1639970e65c2a7e54aeca2e69b1..cd648283db228059066f1898c634f17f96dd8275): Answered review comments.

**Review by phoracek — APPROVED** [review]
/lgtm
/approve

Very nice. Thanks

**Pull request merged**